src/components/organisms/SearchBar.vue

```diff
@@ -24,6 +24,23 @@ export default defineComponent({
     },
   },
   setup() {
+    const params = ref({
+      predicate: null as string | null,
+      distanceMin: null as number | null,
+      distanceMax: null as number | null,
+      instrumentation: null as string | null,
+      dateAndTime: {
+        startDate: null as string | null,
+        endDate: null as string | null,
+        startTime: null as string | null,
+        endTime: null as string | null,
+      },
+    });
```

Comment:
Considering the similarities this structure has with the `parameters` variable in `OtherParams.vue`, it might be cleaner to define these as one or more interfaces (https://www.typescriptlang.org/docs/handbook/interfaces.html). Then, you could just annotate it as
```
const params: Ref<Params> = ....
```
And you could remove all of those `as` statements and typescript will handle everything else.

Reply:
Good call, I'll add that after the drawMap PR is merged to avoid conflict. Changing this to draft in interim.

Reply:
Should be all set